sketch-on-map/src/main/java/com/esri/arcgismaps/sample/sketchonmap/MainActivity.kt

```diff
@@ -0,0 +1,299 @@
+     * polygon button as selected, and enable freehand polygon drawing mode.
+     */
+    fun createModeFreehandPolygon(view: View) {
+        setCurrentSelectionText("FreehandPolygon")
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.freehandPolygonButton.isSelected = true
+        geometryEditor?.apply {
+            tool = freehandTool
+            start(GeometryType.Polygon)
+        }
+    }
+
+    /**
+     * When the undo button is clicked, undo the last event on the GeometryEditor.
+     */
+    fun undo(view: View) {
+        if (geometryEditor?.canUndo?.value == true) {
+            geometryEditor?.undo()
+            setCurrentSelectionText(getString(R.string.undo))
+        }
+    }
+
+    /**
+     * When the redo button is clicked, redo the last undone event on the GeometryEditor.
+     */
+    fun redo(view: View) {
+        if (geometryEditor?.canRedo?.value == true) {
+            geometryEditor?.redo()
+            setCurrentSelectionText(getString(R.string.redo))
+        }
+    }
+
+    /**
+     * When the stop button is clicked, check that sketch is valid. If so, get the geometry from
+     * the sketch, set its symbol and add it to the graphics overlay.
+     */
+    fun stop(view: View) {
+        // get the geometry from sketch editor
+        val sketchGeometry = geometryEditor?.geometry?.value
+            ?: return showError("Error retrieving geometry")
+
+        if (GeometryBuilder.builder(sketchGeometry)?.isSketchValid == false) {
+            return reportNotValid()
+        }
+
+        // stops the editing session
+        geometryEditor?.stop()
+
+        // clear button selection
+        resetButtons()
+
+        // create a graphic from the sketch editor geometry
+        val graphic = Graphic(sketchGeometry)
+        // assign a symbol based on geometry type
+        when (sketchGeometry) {
+            is Polygon -> graphic.symbol = fillSymbol
+            is Polyline -> graphic.symbol = lineSymbol
+            is Point, is Multipoint -> graphic.symbol = pointSymbol
+            else -> {}
+        }
+
+        // add the graphic to the graphics overlay
+        graphicsOverlay.graphics.add(graphic)
+        setCurrentSelectionText("Added graphic to map")
+    }
+
+    /**
+     * De-selects all buttons.
+     */
+    private fun resetButtons() {
+        activityMainBinding.pointLinePolygonToolbar.apply {
+            pointButton.isSelected = false
+            multipointButton.isSelected = false
+            polylineButton.isSelected = false
+            polygonButton.isSelected = false
+            freehandLineButton.isSelected = false
+            freehandPolygonButton.isSelected = false
+        }
+    }
+
+    /**
+     * Clear the MapView of all the graphics and reset selections
+     */
+    fun clear(view: View) {
+        resetButtons()
+        geometryEditor?.clearGeometry()
+        geometryEditor?.clearSelection()
+        setCurrentSelectionText("Cleared")
+    }
```

Comment:
Should this also clear the graphics overlay?

But also I feel there should be a way to throw away the current sketch without saving it onto the graphics overlay but also without clearing the other graphics on the overlay.

Reply:
Also not sure if it makes sense to set the selection text to "cleared" because the geometry editor is still started at this point and there's still a currently-selected tool. Again I think there should be two buttons: one which clears the existing graphics and one which stops the geometry editor without saving the sketch.

Reply:
Added a `restart` button to clear all the committed graphics on the map.